packages/app/src/studio/StudioPanel.vue

```diff
@@ -87,17 +115,31 @@ init({
 onMounted(maybeRenderReactComponent)
 onBeforeUnmount(unmountReactComponent)
 
-loadRemote<StudioApp>('app-studio').then((module) => {
-  if (!module?.default) {
-    error.value = 'The panel was not loaded successfully'
+watch(() => props.studioStatus, (newStatus) => {
+  if (newStatus === 'ENABLED') {
+    loadStudioComponent()
+  }
 
+  maybeRenderReactComponent()
```

Comment:
Under what circumstance is this needed? Wouldn't we always want `loadStudioComponent` to call `maybeRenderReactComponent`?